test/jasmine/tests/calcdata_test.js

```diff
@@ -75,11 +75,11 @@ describe('calculated data and points', function() {
                     categorymode: 'category ascending'
                 }});
 
-                expect(gd.calcdata[0][0].y).toEqual(11);
-                expect(gd.calcdata[0][1].y).toEqual(13);
-                expect(gd.calcdata[0][2].y).toEqual(15);
-                expect(gd.calcdata[0][3].y).toEqual(14);
-                expect(gd.calcdata[0][4].y).toEqual(12);
+                expect(gd.calcdata[0][0]).toEqual(jasmine.objectContaining({x: 2, y: 15}));
+                expect(gd.calcdata[0][1]).toEqual(jasmine.objectContaining({x: 0, y: 11}));
+                expect(gd.calcdata[0][2]).toEqual(jasmine.objectContaining({x: 4, y: 12}));
+                expect(gd.calcdata[0][3]).toEqual(jasmine.objectContaining({x: 1, y: 13}));
+                expect(gd.calcdata[0][4]).toEqual(jasmine.objectContaining({x: 3, y: 14}));
             });
```

Comment:
@etpinard Just an FYI, since I started this CR with the test cases, my conception on where order would be present became outdated. As now I understand that gd.calcdata will keep the trace order, I'm updating test cases such that the original trace order is expected, and explicit checks on the x/y tuples are in place. It looks good to me but please tell me if you have an alternative suggestion.